### PR TITLE
Optimized fibonacci + added average and median

### PR DESCRIPTION
**Pull request by arguiot**
I optimized the Fibonacci algorithm, and I added two statistical algorithms.

### CODE REVIEW

**Comment by mgechev** [comment]
@arguiot would you explain the optimization?

**Comment by arguiot** [comment]
The optimization is mostly in code (fewer lines), it is slightly faster, but not by that much for small numbers

File: src/others/average.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * @param {array} the list that you want the average of.
+ *
+ * @module others/average
+*/
+(function (exports) {
+  'use strict';
+
+  function average(array) {
+    const summed = this.sum(array);
```
Comment (mgechev) [inline src/others/average.js]: Where's `sum` defined?

File: src/others/fibonacci.js [hunk 2]
```diff
@@ -18,22 +18,19 @@
...
-    var n1 = 0;
-    var n2 = 1;
-    var aux;
 
-    while (n > 0) {
-      aux = n1;
-      n1 = n2;
-      n2 += aux;
-      n = n - 1;
+    function recursion(length, originalLength, previous, next) {
```
Comment (mgechev) [inline src/others/fibonacci.js]: This may work a bit faster only with ES2015 and above when there's a TCO.

File: src/others/average.js (hunk 1 above)
Comment (arguiot) [reply src/others/average.js]: I copied from my library DisplayJS, and I forgot to add this 😊, will fix

File: src/others/median.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  'use strict';
+
+  function median(array) {
+    array.sort( (a, b) => a - b );
+    const half = Math.floor(array.length / 2);
+    if(array.length % 2) {
+      return array[half];
+    }
+    else {
+      return (array[half-1] + array[half]) / 2);
```
Comment (mgechev) [inline src/others/median.js]: There's a syntax error.

File: src/others/fibonacci.js (hunk 2 above)
Comment (mgechev) [reply src/others/fibonacci.js]: I'd prefer to not introduce such change, there's no clear performance benefit.

Comment (arguiot) [reply src/others/fibonacci.js]: Yes, but it's still easier to read because there is less code 😊

**Comment by arguiot** [comment]
@mgechev By the way, I didn't touch the tests because I didn't want to break anything 😊

**Pull request closed without merging**